src/btree/bt_cursor.c

```diff
@@ -2441,8 +2569,25 @@ __cursor_range_selectivity(WT_CURSOR_BTREE *start, WT_CURSOR_BTREE *stop, double
         percentile_stop += 0.25 * selectivity_stop_node;
         *selectivityp = percentile_stop - percentile_start;
     } else {
-        /* We reached the same leaf node, so just assume that we match 1/4 of the entries. TODO. */
-        *selectivityp = 0.25 * selectivity_start_node;
+        /* We reached the same leaf. Find the number of matching keys in the leaf's entries  */
+        leaf_count = current_start->page->entries;
```

Comment:
During testing did you happen to see if different leaves have different values for `entries`? I think based on what I was seeing in the tree dumps earlier, `entries` shows the number of actual keys in the leaf page, not the total amount of keys the page is able to store. But just wanted to confirm

Reply:
I think I did see different entries for different leaf nodes 

> total amount of keys the page is able to store

I'm not sure this even makes sense, because keys can have different sizes. So I have totally been assuming it is the number of actual keys in the leaf page.

Reply:
True, makes sense